File: exp-07/q1-fractional-knapsack/solution.py
def main():
    n, w = map(int, input().split())
    items = []
    for _ in range(n):
        weight, profit = map(int, input().split())
        items.append((weight, profit))

    items.sort(key=lambda x: x[1] / x[0], reverse=True)

    total = 0.0
    for weight, profit in items:
        if weight <= w:
            total += profit
            w -= weight
        else:
            total += profit * (w / weight)
            w = 0

    print(f"{total:.2f}")


main()

File: exp-07/q1-fractional-knapsack/solution.js
const fs = require("fs");

const data = fs.readFileSync(0, "utf-8").trim().split(/\s+/).map(Number);
const n = data[0];
let W = data[1];

const items = [];
for (let i = 0; i < n; i++) {
    items.push({ weight: data[2 + 2 * i], profit: data[3 + 2 * i] });
}

items.sort((a, b) => b.profit / b.weight - a.profit / a.weight);

let total = 0;
for (const item of items) {
    if (item.weight <= W) {
        total += item.profit;
        W -= item.weight;
    } else {
        total += item.profit * (W / item.weight);
        W = 0;
    }
}

console.log(total.toFixed(2));